File: java/src/dijkstra/randomTestData.java
package dijkstra;

import java.util.Arrays;
import java.util.Random;

class randomTestData {
	private static final Random rand = new Random(0);

	int n, start;
	int[][] edges;
	int[] expected;

	randomTestData() {
		n = 1 + rand.nextInt(100);
		start = rand.nextInt(n);
		edges = new int[rand.nextInt(n*2)][];
		for (int i = 0; i < edges.length; i++) {
			edges[i] = new int[] { rand.nextInt(n), rand.nextInt(n), 1 + rand.nextInt(100) };
		}

		expected = new int[n];
		Arrays.fill(expected, -1);
		expected[start] = 0;

		for (int count = n-1; count > 0; count--) {
			for (var edge: edges) {
				int u = edge[0], v = edge[1], dist = edge[2];
				if (expected[u] >= 0 && (expected[v] < 0 || expected[u] + dist < expected[v])) {
					expected[v] = expected[u] + dist;
				}
			}
		}
	}
}

File: go/dijkstra/random_test_data.go
package dijkstra

import (
	"math/rand/v2"
)

type randomTestData struct {
	n, start int
	edges    [][]int
	expected []int
}

var random = rand.New(rand.NewPCG(0, 0))

func newRandomTestData() *randomTestData {
	var td randomTestData

	td.n = 1 + random.IntN(100)
	td.start = random.IntN(td.n)
	td.edges = make([][]int, random.IntN(td.n*2))
	for i := range td.edges {
		td.edges[i] = []int{random.IntN(td.n), random.IntN(td.n), 1 + random.IntN(100)}
	}

	td.expected = make([]int, td.n)
	for i := range td.expected {
		td.expected[i] = -1
	}
	td.expected[td.start] = 0

	for count := td.n - 1; count > 0; count-- {
		for _, edge := range td.edges {
			u, v, dist := edge[0], edge[1], edge[2]
			if td.expected[u] >= 0 && (td.expected[v] < 0 || td.expected[u]+dist < td.expected[v]) {
				td.expected[v] = td.expected[u] + dist
			}
		}
	}

	return &td
}
